### PR TITLE
sync bug with errata before get from errata

### PR DESCRIPTION
**Pull request by Ximinhan**
fix issue like https://redhat-internal.slack.com/archives/C04L1FPFHNH/p1729169691264399
use jira api
```
>> res=ErrataConnector()._post("/api/v1/jira/refresh", data=["OCPBUGS-43400"])
ValueError: too many values to unpack (expected 2)

>>> res=ErrataConnector()._post("/api/v1/jira/refresh", json=['OCPBUGS-43400'])
>>> res
<Response [200]>
>>> res.text
'[{"key":"OCPBUGS-43400","status":"Closed","updated":"2024-10-16T23:02:00Z"}]'
```

### CODE REVIEW

File: elliott/elliottlib/bzutil.py [hunk 1]
```diff
@@ -421,6 +422,7 @@ def _get_severity(self):
         return None
 
     def all_advisory_ids(self):
+        errata.sync_jira_issue(self.id)
```
Comment (joepvd) [inline elliott/elliottlib/bzutil.py:425]: Think `all_advisory_ids` gets called pretty often, and it is almost never needed to ask errata to update its list. 

If a jira is attached to an advisory, errata will know. 

If the next version does not ship next week, we will soon not check: 

Remains the case where we prepare 4.y and 4.(y+1), and we prepare 4.y earlier than 4.(y+1). 

Suggesting the optimization to first get `all_advisory_ids`, then if empty or error, sync and try again.

File: elliott/elliottlib/bzutil.py [hunk 2]
```diff
@@ -214,6 +214,7 @@ def is_invalid_tracker_bug(self):
         return has_keywords or has_cve_in_summary
 
     def all_advisory_ids(self):
+        errata.sync_bugzilla_bug(self.id)
```
Comment (joepvd) [inline elliott/elliottlib/bzutil.py:217]: Think this is not needed. We only have flaw bugs in bugzilla, and these do not get this treatment.

File: elliott/elliottlib/bzutil.py (hunk 1 above)
Comment (Ximinhan) [reply elliott/elliottlib/bzutil.py]: updated in https://github.com/openshift-eng/art-tools/pull/1063/commits/2e297444788301fb704f7cf0fcad88cceccfaef0

File: elliott/elliottlib/bzutil.py (hunk 2 above)
Comment (Ximinhan) [reply elliott/elliottlib/bzutil.py]: updated in https://github.com/openshift-eng/art-tools/pull/1063/commits/2e297444788301fb704f7cf0fcad88cceccfaef0

**Comment by joepvd** [comment]
```
Result not handled: {"error":"JIRA issue OCPBUGS-43335 not found."}
While fetching: https://errata.devel.redhat.com/jira_issues/OCPBUGS-43335/advisories.json
Result not handled: {"error":"JIRA issue OCPBUGS-34277 not found."}
While fetching: https://errata.devel.redhat.com/jira_issues/OCPBUGS-34277/advisories.json
Result not handled: {"error":"JIRA issue OCPBUGS-43450 not found."}
While fetching: https://errata.devel.redhat.com/jira_issues/OCPBUGS-43450/advisories.json
Result not handled: {"error":"JIRA issue OCPBUGS-43443 not found."}
While fetching: https://errata.devel.redhat.com/jira_issues/OCPBUGS-43443/advisories.json
Found the following problems, please investigate
Failed to get advisories for bug OCPBUGS-43335: too many values to unpack (expected 2)
Failed to get advisories for bug OCPBUGS-34277: too many values to unpack (expected 2)
Failed to get advisories for bug OCPBUGS-43450: too many values to unpack (expected 2)
Failed to get advisories for bug OCPBUGS-43443: too many values to unpack (expected 2)
```
In https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Fpromote-assembly/395/console

File: elliott/elliottlib/cli/verify_attached_bugs_cli.py [hunk 3]
```diff
@@ -480,10 +483,13 @@ def _verify_blocking_bugs(self, blocking_bugs_for, is_attached=False):
...
                         blocker_advisories = blocker.all_advisory_ids()
                     except ErrataException as e:
-                        message = f"Failed to get advisories for bug {blocker.id}: {e}"
-                        logger.error(message)
-                        self._complain(message)
-                        continue
+                        try:
+                            sync_jira_issue(blocker.id)
+                        except Exception as e:
+                            message = f"Failed to get advisories for bug {blocker.id}: {e}"
```
Comment (thegreyd) [inline elliott/elliottlib/cli/verify_attached_bugs_cli.py:489]:
```suggestion
                            message = f"Failed to sync bug {blocker.id}: {e}"
```

File: elliott/elliottlib/cli/verify_attached_bugs_cli.py [hunk 4]
```diff
@@ -249,7 +249,10 @@ async def get_all_advisory_ids(bug):
             try:
                 all_advisories_id = bug.all_advisory_ids()
             except ErrataException as e:
-                return f'Failed to get advisories for bug {bug.id}: {e}'
+                try:
+                    sync_jira_issue(bug.id)
+                except Exception as e:
+                    return f'Failed to get advisories for bug {bug.id}: {e}'
```
Comment (thegreyd) [inline elliott/elliottlib/cli/verify_attached_bugs_cli.py:255]:
```suggestion
                    return f'Failed to sync bug {bug.id}: {e}'
```

File: elliott/elliottlib/errata.py [hunk 5]
```diff
@@ -111,6 +111,14 @@ def add_jira_issue(advisory_id, jira_issue_id):
     return ErrataConnector()._post(f"/api/v1/erratum/{advisory_id}/add_jira_issue", json={'jira_issue': jira_issue_id})
 
 
+def sync_jira_issue(jira_issue_id):
+    """
+    Sync a jira issue to advisory
+    Response code will return
+    """
+    return ErrataConnector()._post("/api/v1/jra/refresh", data=[jira_issue_id])
```
Comment (thegreyd) [inline elliott/elliottlib/errata.py:119]:
```suggestion
    return ErrataConnector()._post("/api/v1/jra/refresh", json=[jira_issue_id])
```

**Pull request merged**